### PR TITLE
feature: Creates Splash Screen

### LINKED ISSUE #3: [US01] Visualizar a apresentação inicial do aplicativo
### Descrição
Tela que é exibida ao abrir o aplicativo.

### Protótipo

<p float="left">
  splash screen 1
  splash screen 2
</p>

<p float="left">
  splash screen 3
  início
</p>

### Regra de Negócio
- Deverá seguir o layout do protótipo
- Se for o primeiro acesso do usuário, as splash screens deverão ser exibidas
- Ao clicar no link **Continuar**, deverá ser redirecionado para a tela seguinte
- Ao clicar no link **Pular**, deverá ser redirecionado para a tela de **Início**
- Ao clicar no botão **Vamos começar!**, deverá ser redirecionado para a tela de **Início**
- Ao clicar no botão **Login**, deverá ser redirecionado para a tela de **Login**
- Ao clicar no botão **Cadastro**, deverá ser redirecionado para a tela de **Cadastro**

### Critérios de Aceite
- [ ] O layout está de acordo com o protótipo 
- [ ] Ao acessar pela primeira vez, o usuário visualiza as splash screens
- [ ] Ao clicar no link **Continuar**, o usuário é redirecionado para a tela seguinte
- [ ] Ao clicar no link **Pular**, o usuário é redirecionado para tela **Início**
- [ ] Ao clicar no botão **Vamos começar!**, o usuário é redirecionado para a tela de **Início**
- [ ] Ao clicar no botão **Login** da tela **Início**, o usuário é redirecionado para a tela de **Login**
- [ ] Ao clicar no botão **Cadastro** da tela **Início**, o usuário é redirecionado para a tela de **Cadastro**

### PR DESCRIPTION
**Pull request by larissa-sales**
### Description

Creates the app onboarding screens.

### Related Issues
Closes #3 

### Screeshots

<p float="left">
  splash1
  splash2
splash2
</p>

### CODE REVIEW

File: src/routes/navigation/rootNavigator.tsx [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+import { NavigationContainer, StackActions } from '@react-navigation/native';
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
 import TabNavigator from "./TabNavigator";
+import SplashView from "../../views/SplashView";
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+const Stack = createNativeStackNavigator();
 
 const RootNavigator = (): React.ReactElement => {
+	const [isAppFirstLaunch, setIsAppFirstLaunch] = useState(null);
```
Comment (lucianaribeiro) [inline src/routes/navigation/rootNavigator.tsx:11]: Should fix indentation with prettier

File: src/routes/navigation/rootNavigator.tsx [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
+		const fetchData = async () => {
+			const appData = await AsyncStorage.getItem("isAppFirstLaunch");
+			if(appData == null) { 
+				setIsAppFirstLaunch(true);
+				AsyncStorage.setItem("isAppFirstLaunch", 'false');
+			}
+			else { 
+				setIsAppFirstLaunch(false)
+			}
+		}
```
Comment (lucianaribeiro) [inline src/routes/navigation/rootNavigator.tsx:23]: same here

File: src/routes/navigation/rootNavigator.tsx [hunk 3]
```diff
@@ -1,14 +1,39 @@
...
-        </NavigationContainer>
-    );
+	return (
+		isAppFirstLaunch != null && (
+			<NavigationContainer>
+				<Stack.Navigator screenOptions={{ headerShown: false }}>
+					{isAppFirstLaunch && ( <Stack.Screen name="SplashScreen" component={SplashView} />)}
+					 <Stack.Screen name="TabNavigator" component={TabNavigator} />
+				</Stack.Navigator>
+			</NavigationContainer>
```
Comment (lucianaribeiro) [inline src/routes/navigation/rootNavigator.tsx:34]: same here

File: src/views/SplashView/index.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        <Text style={styles.nextLink}>{Labels.splashScreenNext}</Text>
+      </View>
+    );
+  };
+
+  const renderDoneButton = () => {
+    return (
+			<View>
+				<Text style={styles.doneBtn}>{Labels.splashScreenDone}</Text>
+      </View>
```
Comment (lucianaribeiro) [inline src/views/SplashView/index.tsx:58]: omg this indentation

**Review by lucianaribeiro — CHANGES_REQUESTED** [review]
Please fix the indentation problems installing prettier to VSCode

File: src/routes/navigation/rootNavigator.tsx [hunk 5]
```diff
@@ -1,14 +1,47 @@
...
+const SplashScreenStack = createNativeStackNavigator();
 
 const RootNavigator = (): React.ReactElement => {
+  const [isAppFirstLaunch, setIsAppFirstLaunch] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      // AsyncStorage.clear();
+      const appData = await AsyncStorage.getItem("isAppFirstLaunch");
+      if (appData == null) {
```
Comment (lucianaribeiro) [inline src/routes/navigation/rootNavigator.tsx:17]: `AppData` is not a significant name, would be better something like `isAppFirstLaunch`

Comment (lucianaribeiro) [reply src/routes/navigation/rootNavigator.tsx:17]: Use triple `===` instead of only two

File: src/routes/navigation/rootNavigator.tsx [hunk 6]
```diff
@@ -1,14 +1,47 @@
...
+import { NavigationContainer } from "@react-navigation/native";
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
 import TabNavigator from "./TabNavigator";
+import SplashView from "../../views/SplashView";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+const SplashScreenStack = createNativeStackNavigator();
 
 const RootNavigator = (): React.ReactElement => {
+  const [isAppFirstLaunch, setIsAppFirstLaunch] = useState(null);
```
Comment (lucianaribeiro) [inline src/routes/navigation/rootNavigator.tsx:11]: Maybe would be better to user some other name like `splashScreen` `setSplashScreen`

File: src/routes/navigation/rootNavigator.tsx [hunk 7]
```diff
@@ -1,14 +1,47 @@
...
+    fetchData().catch(console.error);
+  }, []);
 
-    return (
-        <NavigationContainer>
-            <TabNavigator />
-        </NavigationContainer>
-    );
+  return (
+    isAppFirstLaunch != null && (
```
Comment (lucianaribeiro) [inline src/routes/navigation/rootNavigator.tsx:28]: I don't think you nesse the `!= null` validator, `isAppFirstLaunch && (` should handle it

Comment (larissa-sales) [reply src/routes/navigation/rootNavigator.tsx:28]: i tested, but didn't work, so i had to leave it that way

**Review by lucianaribeiro — APPROVED** [review]

**Comment by larissa-sales** [comment]
A new PR was open because we were unnable to rebase this one. See PR https://github.com/Plant-me-app/plant.me-app/pull/45

**Pull request closed without merging**